### PR TITLE
Add eos_ospf_interfaces resource module

### PR DESCRIPTION
**Pull request by GomathiselviS**
##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME
eos_ospf_interfaces
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: plugins/module_utils/network/eos/facts/ospf_interfaces/ospf_interfaces.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+
+class Ospf_interfacesFacts(object):
+    """ The eos ospf_interfaces facts class
+    """
+
+    def __init__(self, module, subspec="config", options="options"):
+        self._module = module
+        self.argument_spec = Ospf_interfacesArgs.argument_spec
+        spec = deepcopy(self.argument_spec)
```
Comment (NilashishC) [inline plugins/module_utils/network/eos/facts/ospf_interfaces/ospf_interfaces.py:38]: I _think_ this section of code (38-47) is not required. I will remove it from builder template. But for now, removing it from here will bump code cov too.

File: tests/integration/targets/eos_ospf_interfaces/tests/common/merged.yaml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    - become: true
+      arista.eos.eos_facts:
+        gather_network_resources: ospf_interfaces
+
+    - assert:
+        that:
+          - result.commands|length == 16
+          - result.changed == true
+          - result.commands|symmetric_difference(merged.commands) == []
+          - result.after|symmetric_difference(ansible_facts['network_resources']['ospf_interfaces']) == []
```
Comment (NilashishC) [inline tests/integration/targets/eos_ospf_interfaces/tests/common/merged.yaml:54]: Asserting before/after from gathered facts is not much useful, imo. Both are generated from the same code path.

File: meta/runtime.yml [hunk 3]
```diff
@@ -97,6 +97,9 @@ plugin_routing:
     eos_ospfv3:
       redirect: arista.eos.eos
     ospfv3:
+    eos_ospf_interfaces:
+      redirect: arista.eos.eos
+    ospf_interfaces:
       redirect: arista.eos.eos
     eos_static_route:
```
Comment (NilashishC) [inline meta/runtime.yml:104]: Entry to make ospf_interfaces -> eos_ospf_interfaces is missing in `modules` dict.

**Review by NilashishC — CHANGES_REQUESTED** [review]

File: tests/integration/targets/eos_ospf_interfaces/tests/common/merged.yaml (hunk 2 above)
Comment (GomathiselviS) [reply tests/integration/targets/eos_ospf_interfaces/tests/common/merged.yaml:54]: I might have to change this assertion for most of the interfaces related tests. I shall take it up as a separate PR.

File: meta/runtime.yml [hunk 4]
```diff
@@ -206,6 +207,8 @@ plugin_routing:
       redirect: arista.eos.eos_ospfv2
     ospfv3:
       redirect: arista.eos.eos_ospfv3
+    ospf_interfaces:
+          redirect: arista.eos.eos_ospf_interfaces
```
Comment (NilashishC) [inline meta/runtime.yml:211]: There maybe an indentation issue here?

**Review by NilashishC — APPROVED** [review]

**Pull request merged**